submissions/iakow/js-dom/main.js

```diff
@@ -0,0 +1,169 @@
+      "As a new team at a time when there were more F1 cars than spaces on the grid, Jordan initially had to pre-qualify, but the 191 made that a relatively easy task for regular drivers Andrea de Cesaris and Bertrand Gachot. Jordan scored its first points in the Canadian GP and the team ended up fifth in the constructors' table, behind only big hitters McLaren, Williams, Ferrari and Benetton.",
+      "The 191 also helped Michael Schumacher launch his F1 career, with a stunning performance at Spa, and de Cesaris might have won the Belgian GP had it not been for an engine failure, the result of a miscommunication between engine supplier Cosworth and the team."
+    ]
+  },
+  {
+    name: "Lotus 79",
+    years: "1978", design: "Peter Wright, Colin Chapman, Geoff Aldridge, Martin Ogilvie, Tony Rudd",
+    wins: "6",
+    pic: "./img/Lotus_79.jpg",
+    descr: [
+      "Ground-breaking design? Check. Iconic livery? Check. Uncluttered lines? Check. Beautiful engineering? Check.",
+      "The Lotus 79, which refined the ground-effect concept and dominated the 1978 F1 season, ticks all the boxes for this list. It's also even more impressive close up in the flesh than it is in the famous period images of Mario Andretti and Ronnie Peterson.",
+      "Lower, neater and sleeker than the Lotus 78 that preceded it, the 79 won on its debut in the Belgian GP in the hands of Andretti and finished 1-2 next time out in Spain. It was only beaten to pole once during the campaign, though the odd reliability issue and jump-start penalty for Andretti at Monza meant it only won six of the 11 championship rounds it started in 1978.",
+      "By 1979 the opposition had caught up in the ground-effect stakes and the Lotus 79 was limited to four podiums. Even then it looked impressive in its green Martini livery, but it's the JPS-liveried original that deservedly earns our top spot."
+    ]
+  },
+]
+
+const carPhoto = document.querySelector('.photo');
+const carName = document.querySelector('h2');
+const carYears = document.querySelector('.years');
+const carDesign = document.querySelector('.design');
+const carWins = document.querySelector('.wins');
+const carDescription = document.querySelector('.descr');
+
+let currentCarIndex = "0";
+
+const handler = (e) => {
+  const navLink = e.target;
+
+  if (navLink.id === currentCarIndex) return;
+  currentCarIndex = navLink.id;
+
+  document.querySelector('.active').classList.toggle('active');
+  navLink.classList.toggle('active');
+
+  carPhoto.setAttribute('src', CARS[navLink.id].pic);
+  carName.innerText = CARS[navLink.id].name;
+  carYears.innerText = CARS[navLink.id].years;
+  carDesign.innerText = CARS[navLink.id].design;
+  carWins.innerText = CARS[navLink.id].wins;
+
+  document.querySelectorAll('.descr > p').forEach(item => item.remove());
```

Comment:
You don't need to remove these nodes from your DOM. Just use `innerHTML` for inserting a new content there. It'll overwrite the previous content inside this element.

Reply:
There are different numbers of paragraphs in different cases. So I can't just inject new content into paragraphs.

Or did you mean I should use innerHTML for the parent? But this way I still delete these nodes. (Or not?). Аnd this option will complicate the logic.

I'm not sure if I understand you correctly. Please clarify.

Reply:
I mean you can use the `innerHTML` when you insert content for description into the `carDescription` node.
You can remove `carDescription.append(p);` (line 161) and rebuilt your logic with the descriptions using `Document.createDocumentFragment()`. More info [here](https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment).
It will help you to get rid of the line 156 and improve performance a little bit with adding description nodes.
Let me know if it's clear for you

Reply:
Got it, thanks!